components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -180,6 +185,26 @@ class SitePermissionsFeature(
                     buttons
                 )
             }
+            is ContentVideoCapture -> {
+                createPromptForCameraPermission(
+                    context,
+                    uriString,
+                    permissionRequest,
+                    session,
+                    allowButtonTitle,
+                    denyButton
+                )
+            }
+            is ContentVideoCamera -> {
+                createPromptForCameraPermission(
+                    context,
+                    uriString,
+                    permissionRequest,
+                    session,
+                    allowButtonTitle,
+                    denyButton
+                )
```

Comment:
Is this exactly the same? Could this be reduced to `is ContentVideoCapture or is ContentVideoCamera` or something like that?

Reply:
I thought the same thing, but I couldn't find the right syntax to make it work, I've tried multiple alternatives without any luck. Any suggestions? :)



![image](https://user-images.githubusercontent.com/773158/53095916-f2faea80-34eb-11e9-9a24-7e805639e62d.png)

![image](https://user-images.githubusercontent.com/773158/53096162-80d6d580-34ec-11e9-8774-47423bb7fb2e.png)

![image](https://user-images.githubusercontent.com/773158/53096075-440ade80-34ec-11e9-95fb-d4c348dfedd4.png)

Reply:
Oh, okay :(

Reply:
The logic says it should be right to do it, because in a java `switch` is possible, but maybe it has something to do with compiler want a explicit declaration per branch.